src/store/fileUploadStore.js

```diff
@@ -349,7 +349,20 @@ const actions = {
 				context.commit('markFileAsPendingUpload', { uploadId, index, sharePath: uniquePath })
 			} catch (exception) {
 				console.error(`Error while uploading file "${fileName}":` + exception.message, fileName)
-				showError(t('spreed', 'Error while uploading file "{fileName}"', { fileName }))
+				if (exception.response) {
+					try {
+						const responseText = await exception.response.text()
+						const parser = new DOMParser()
+						const xmlDoc = parser.parseFromString(responseText, 'application/xml')
+						const messageElement = xmlDoc.getElementsByTagName('s:message')[0]
+						if (messageElement) {
+							const errorMessage = messageElement.textContent
```

Comment:
This part parses a specific XML response and should be a part of the transport application layer, not a global UI state module (store). Let's move this logic out from Vuex. So in the UI part we know that the process was not successful and there is a specific error.

Also, we can consider it as a duplicated (= unavailable) name and proceed to uploading with a different name.